test/biggystring.test.ts

```diff
@@ -199,12 +199,33 @@ describe('toBns', function () {
   it('regular number', function () {
     assert.equal(toBns(123), '123')
   })
-  it('scientific notation', function () {
-    assert.equal(toBns(12e11), '1200000000000')
-  })
-  it('resolve negative numbers', function () {
-    assert.equal(toBns(-3e10), '-30000000000')
-    assert.equal(toBns(-3e-10), '-0.0000000003')
+  it('scientific notation: big numbers', function () {
+    const big = 3000000000000000000000
+    const bigStr = big.toString()
+    assert.equal(bigStr, '3e+21')
+    assert.equal(toBns(big), '3000000000000000000000')
+    assert.equal(toBns(bigStr), '3000000000000000000000')
+  })
+  it('scientific notation: small numbers', function () {
+    const small = 0.000000000000000000003
+    const smallStr = small.toString()
+    assert.equal(smallStr, '3e-21')
+    assert.equal(toBns(small), '0.000000000000000000003')
+    assert.equal(toBns(smallStr), '0.000000000000000000003')
+  })
+  it('scientific notation: big negative numbers', function () {
```

Comment:
I still feel like this needs more tests. 

Checking decimal values like `3.1234e12` `.001e12` `0.000123e12`

invalid values like `3.1.0e123` `3x0e12` `